sys/uio.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * sys/uio.c
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Ziemowit Leszczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <unistd.h>
+#include <limits.h>
+#include <sys/uio.h>
+
+
+static int checkv(const struct iovec *iov, int iovcnt)
+{
+	ssize_t max_len = SSIZE_MAX;
+
+	if (iovcnt <= 0 || iovcnt > IOV_MAX)
+		return -1;
+
+	for (unsigned int i = 0; i < iovcnt; ++i) {
```

Comment:
"Local variables

Local variables should be defined before the function code according to ANSI C 89 standard. (...)"

https://phoenix-rtos.com/documentation/coding.md